zeebe/qa/integration-tests/src/test/java/io/camunda/zeebe/it/clustering/ClusterStatusTest.java

```diff
@@ -0,0 +1,123 @@
+import io.camunda.zeebe.qa.util.cluster.TestCluster;
+import io.camunda.zeebe.qa.util.cluster.TestStandaloneBroker;
+import io.camunda.zeebe.qa.util.junit.ZeebeIntegration;
+import io.camunda.zeebe.qa.util.junit.ZeebeIntegration.TestZeebe;
+import java.time.Duration;
+import org.awaitility.Awaitility;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.testcontainers.junit.jupiter.Testcontainers;
+
+/**
+ * Integration test to verify cluster status reporting.
+ *
+ * <p>Tests that the cluster status correctly reports healthy when at least one partition has a
+ * healthy leader, and unhealthy when no partitions have healthy leaders.
+ */
+@Testcontainers
+@ZeebeIntegration
+public class ClusterStatusTest {
+
+  @TestZeebe(autoStart = false)
+  private static final TestCluster CLUSTER =
+      TestCluster.builder()
+          .withEmbeddedGateway(false)
+          .withGatewaysCount(1)
+          .withGatewayConfig(gateway -> gateway.withCreateSchema(false))
+          .withBrokersCount(3)
+          .withBrokerConfig(node -> node.withCreateSchema(false))
+          .withPartitionsCount(3)
+          .withReplicationFactor(3)
+          .build();
+
+  @BeforeEach
+  void beforeEach() {
+    CLUSTER.start().awaitCompleteTopology();
+  }
+
+  @Test
+  public void shouldReportHealthyStatusWhenClusterIsUp() {
+    // given - cluster is running with leaders
+
+    // when - requesting status
+    try (final var client = createCamundaClient()) {
+      final var status = client.newStatusRequest().send().join();
+
+      // then - status should be healthy
+      assertThat(status.isHealthy()).isTrue();
+    }
+  }
+
+  @Test
+  public void shouldReportUnhealthyStatusWhenNoLeadersExist() {
+    // given - cluster is initially healthy
+    try (final var client = createCamundaClient()) {
+      final var initialResponse = client.newStatusRequest().send().join();
+      assertThat(initialResponse.isHealthy()).isTrue();
+
+      // when - stopping all brokers to force no leaders for any partition
+      CLUSTER.brokers().values().forEach(TestStandaloneBroker::stop);
+
+      // then - status should eventually become unhealthy
+      Awaitility.await("Status should become unhealthy when no leaders exist")
+          .atMost(Duration.ofSeconds(30))
+          .pollInterval(Duration.ofSeconds(1))
+          .untilAsserted(
+              () -> {
+                final var status = client.newStatusRequest().send().join();
+                assertThat(status.isHealthy()).isFalse();
+              });
+    }
+  }
+
+  @Test
+  public void shouldReportHealthyStatusWhenOnlyOnePartitionHasHealthyLeader() {
+    // given - cluster is initially healthy
+    try (final var client = createCamundaClient()) {
+      final var initialResponse = client.newStatusRequest().send().join();
+      assertThat(initialResponse.isHealthy()).isTrue();
+
+      // when - stopping 2 out of 3 brokers, leaving at least one partition with a leader
```

Comment:
:x: Since replication factor is 3, if you stop 2 brokers then there won't be a leader for any partition. The test is likely passing now because the update that leader does not exist is not received yet by the gateway.

For this test to work, I would suggest to set the replicationFactor to 1. For the other two tests it does not matter what the replicationFactor is. 

To ensure we are checking the status correctly, may be wait until two partitions has no leader before checking the status is still healthy.